torch/_dynamo/utils.py

```diff
@@ -3595,4 +3595,8 @@ def set_feature_use(feature: str, usage: bool):
     Records whether we are using a feature
     Generally a feature is a JK.
     """
-    get_metrics_context().set_key_value("feature_usage", feature, usage)
+    try:
```

Comment:
I don't like it. Seems like a good way to mask problems. At the very least we can do this:

```
if metrics_context.in_progress():
    metrics_context.set_key_value(...)
```

But stick with me here, I'm not trying to blow everything up just yet... I'm wondering if it makes more sense to centralize the logging of all features right before we finalize the record that we log to scuba. We have such a location. Or is it the case that we'd rather only capture the feature usage when that code path is actually executed and otherwise omit it from the logs?

Reply:
So we can just record the feature config value right before the log, but we already store the config :). My thought is we want to record "usage" - i.e. we actually did a write / read from inductor fx cache. Let me know if that's a hard blocker, and I can just double check we record the relevant configs (if not fix them), and then mark these tasks as redundant.

Reply:
Ok, if that's what you're aiming at, then let's do my other suggestion of checking `in_progress`?

Reply:
Done.